atc/lidar/scanner.go

```diff
@@ -107,7 +90,15 @@ func (s *scanner) check(ctx context.Context, checkable db.Checkable, resourceTyp
 		return
 	}
 
-	_, created, err := s.checkFactory.TryCreateCheck(lagerctx.NewContext(spanCtx, logger), checkable, resourceTypes, version, false)
+	// We don't want to pass in a nil list of resource types here if the pipeline
+	// does not have any resource types because we will be calling methods off of
+	// the db.ResourceTypes interface in the TryCreateCheck method
+	var resourceTypes db.ResourceTypes
+	if rts, found := resourceTypesMap[checkable.PipelineID()]; found {
+		resourceTypes = rts
+	}
+
```

Comment:
Oh sorry I meant a nil `db.ResourceTypes` rather than passing a `nil` to the `TryCreateCheck` method. Does that make sense? D:

Reply:
Still a bit confused tbh. `db.ResourceTypes` is just a `[]db.ResourceType` (not an interface), so `nil` should be fine, right? I don't see how this code is different from just:

```go
resourceTypes := resourceTypesMap[checkable.PipelineID()]
```

...since if it's not found, it'll just take the zero-value (a `nil` slice) anyway

Reply:
Ohh my bad I thought that if you don't find a key in a map it will return a `nil` and not a nil slice. But I'm wrong! Looks like if you don't find a key in a map, it just returns the default nil value which is a nil slice of that type. I just wanted to make sure that it was never a real `nil` value being passed into `TryCreateCheck` because the `db.ResourceTypes` is being used as an interface in that method. Thanks for pointing that out haha